### ISSUE TITLE
"Mismatch between m_dSourceCenterLon and nA" when generating map from CS DGLL to RLL

### ISSUE THREAD

**Issue by simonbyrne** [ClimateGlobalChange/tempestremap#98]
```
ne=6
nlon=30
nlat=30

./GenerateCSMesh --res $ne --alt --file mesh_cs.g
./GenerateRLLMesh --lon $nlon --lat $nlat --file mesh_rll.g
./GenerateOverlapMesh --a mesh_cs.g --b mesh_rll.g --out mesh_overlap.g
./GenerateOfflineMap --in_mesh mesh_cs.g --out_mesh mesh_rll.g --ov_mesh mesh_overlap.g --in_type dgll --in_np 4 --out_map weights.nc
```

The last line gives the following error:
```
├ ./GenerateOfflineMap --in_mesh mesh_cs.g --out_mesh mesh_rll.g --ov_mesh mesh_overlap.g --in_type dgll --in_np 4 --out_map weights.nc
Parameters:
  --in_mesh <string> ["mesh_cs.g"]
  --out_mesh <string> ["mesh_rll.g"]
  --ov_mesh <string> ["mesh_overlap.g"]
  --in_type <string> ["dgll"] [fv|cgll|dgll]
  --out_type <string> ["fv"] [fv|cgll|dgll]
  --out_map <string> ["weights.nc"]
  --in_meta <string> [""]
  --out_meta <string> [""]
  --in_concave <bool> [false]
  --out_concave <bool> [false]
  --in_np <integer> [4]
  --out_np <integer> [4]
  --method <string> [""]
  --mono <bool> [false]
  --nobubble <bool> [false]
  --nocorrectareas <bool> [false]
  --noconserve <bool> [false]
  --nocheck <bool> [false]
  --sparse_constraints <bool> [false]
  --in_data <string> [""]
  --out_data <string> [""]
  --var <string> [""]
  --ncol_name <string> ["ncol"]
  --out_double <bool> [false]
  --preserve <string> [""]
  --preserveall <bool> [false]
  --fillvalue <double> [0.000000]
  --out_format <string> ["Netcdf4"] [Classic|Offset64Bits|Netcdf4|Netcdf4Classic]
------------------------------------------------------------
Initializing dimensions of map
..Input mesh
..Output mesh
Loading input mesh
..Mesh size: Nodes [218] Elements [216]
Loading output mesh
..Mesh size: Nodes [872] Elements [900]
Loading overlap mesh
..Mesh size: Nodes [1834] Elements [1872]
..Calculating input mesh Face areas
....Input Mesh Geometric Area: 1.256637061435796e+01 (9.999999999999037e-01)
..Calculating output mesh Face areas
....Output Mesh Geometric Area: 1.256637061435917e+01 (9.999999999999996e-01)
..Overlap mesh forward correspondence found
Calculating overlap mesh Face areas
..Overlap Mesh Area: 1.256637061435917e+01 (1.000000000000000e+00)
Correcting source/target areas to overlap mesh areas
Generating input mesh meta data
..Input Mesh Numerical Area: 1.256637061435919e+01
WARNING: Significant mismatch between input mesh numerical area and geometric area
Calculating offline map
..Element 0/216
..Map generation complete
------------------------------------------------------------
Analyzing map
..Per-dof consistency  (tol 1.00000e-08).. PASS
..Per-dof conservation (tol 1.00000e-08).. PASS
..Weights within range [-10,+10].. Done
..
..  Total nonzero entries: 29952
..   Column index min/max: 1 / 3456 (3456 source dofs)
..      Row index min/max: 1 / 900 (900 target dofs)
..      Source area / 4pi: 1.000000000000003e+00
..    Source area min/max: 3.069455126572463e-04 / 1.189831970791136e-02
..      Target area / 4pi: 9.999999999999973e-01
..    Target area min/max: 1.139030049808762e-03 / 2.197188441211793e-02
..    Source frac min/max: 9.999999999999345e-01 / 1.000000000000055e+00
..    Target frac min/max: 9.999999999999992e-01 / 1.000000000000001e+00
..    Map weights min/max: -1.518365879622196e-01 / 8.043545733580668e-01
..       Row sums min/max: 9.999999999999992e-01 / 1.000000000000001e+00
..   Consist. err min/max: -7.771561172376096e-16 / 8.881784197001252e-16
..    Col wt.sums min/max: 9.999999999999346e-01 / 1.000000000000055e+00
..   Conserv. err min/max: -6.539213615042172e-14 / 5.506706202140776e-14
..    Global conserv. err: 3.730349362740526e-14
..
..Histogram of nonzero entries in sparse matrix
....Column 1: Number of nonzero entries (bin minimum)
....Column 2: Number of columns with that many nonzero values
....Column 3: Number of rows with that many nonzero values
..[[5, 128, 0],[6, 1152, 0],[7, 512, 0],[8, 768, 0],[9, 128, 0],[10, 128, 0],[11, 128, 0],[12, 128, 0],[14, 128, 0],[16, 128, 280],[27, 128, 0],[31, 0, 620]]
..
..Histogram of weights
....Column 1: Lower bound on weights
....Column 2: Upper bound on weights
....Column 3: # of weights in that bin
..[[-1, 0, 12432],[0, 1, 17520]]
------------------------------------------------------------
Writing offline map
..EXCEPTION (../src/OfflineMap.cpp, Line 2342) Mismatch between m_dSourceCenterLon and nA
```

**Comment by jakebolewski** [comment]
Does it work if you increase the resolution of the RLL grid?

**Comment by simonbyrne** [comment]
no.

**Comment by jakebolewski** [comment]
Looking in the debugger the  number of face areas doesn't match up with the center long values from the source mesh.

**Comment by paullric** [comment]
Fixed in commit 8f7524b43a389ae8e8408e8cefb65ecce6d4c0a4
